src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs

```diff
@@ -29,6 +31,57 @@ public void AddExternalDirective_EnsureAvailableInSchema()
             Assert.Equal(0, directive.Arguments.Count());
             Assert.Collection(directive.Locations,
                 t => Assert.Equal(DirectiveLocation.FieldDefinition, t));
-        }        
+        }
+
+        [Fact]
+        public void AnnotateExternalToObjectFieldCodeFirst()
+        {
+            // arrange
+            // act
+            var schema = Schema.Create(
+                t =>
+                {
+                    t.RegisterQueryType(new ObjectType(
+                        o => o.Name("Query")
+                            .Field("field")
+                            .Argument("a", a => a.Type<StringType>())
+                            .Type<StringType>()
+                            .External()
+                    ));
+
+                    t.RegisterDirective<ExternalDirectiveType>();
+
+                    t.Use(next => context => default(ValueTask));
+                });
+
+            ObjectType query = schema.GetType<ObjectType>("Query");
+            query.Fields["field"].Directives.Single(t => t.Name == "external");
+        }
+
+        [Fact]
+        public void AnnotateExternalToInterfaceFieldSchemaFirst()
+        {
+            // arrange
+            // act
+            var schema = SchemaBuilder.New()
+                .AddDocumentFromString(
+                    @"
+                    type Query {
+                        field(a: Int): String
+                            @external
+                    }
+
+                    interface IQuery {
+                        field(a: Int): String
+                            @external
+                    }
+                    ")
+                .AddDirectiveType<ExternalDirectiveType>()
+                .Use(next => context => default(ValueTask))
+                .Create();
+
+            InterfaceType queryInterface = schema.GetType<InterfaceType>("IQuery");
```

Comment:
where is the assert part of this test?

Reply:
Replaced the collectin.single with Assert.Collection